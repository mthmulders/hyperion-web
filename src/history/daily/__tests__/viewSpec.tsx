import * as React from 'react';

import { shallow } from 'enzyme';

import View from '../view';

describe('<View />', () => {
    it('should present some data', () => {
        // Arrange
        const data = {
            elecCon: 0.265,
            electricityLow: 2150.714,
            electricityNormal: 2702.641,
            gas: 3781.676,
            recordDate: new Date('2018-01-05'),
        };

        // Act
        const wrapper = shallow(<View data={ data } />);

        // Assert
        expect(wrapper).toMatchSnapshot();
    });
});
