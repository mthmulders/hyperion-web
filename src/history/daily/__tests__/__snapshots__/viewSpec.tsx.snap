// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<View /> should present some data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DailyReadingView
    data={
        Object {
            "elecCon": 0.265,
            "electricityLow": 2150.714,
            "electricityNormal": 2702.641,
            "gas": 3781.676,
            "recordDate": 2018-01-05T00:00:00.000Z,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "borderless": true,
      "children": <tbody>
        <tr>
                <th
                        scope="row"
                >
                        Date recorded
                </th>
                <td>
                        Friday, January 5th 2018
                </td>
        </tr>
        <tr>
                <th
                        scope="row"
                >
                        Electricity (low)
                </th>
                <td>
                        2150.714
                         kWh
                </td>
        </tr>
        <tr>
                <th
                        scope="row"
                >
                        Electricity (normal)
                </th>
                <td>
                        2702.641
                         kWh
                </td>
        </tr>
        <tr>
                <th
                        scope="row"
                >
                        Gas
                </th>
                <td>
                        3781.676
                         m
                        <sup>
                                3
                        </sup>
                </td>
        </tr>
</tbody>,
      "responsive": true,
      "responsiveTag": "div",
      "tag": "table",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <tr>
            <th
                        scope="row"
            >
                        Date recorded
            </th>
            <td>
                        Friday, January 5th 2018
            </td>
</tr>,
          <tr>
            <th
                        scope="row"
            >
                        Electricity (low)
            </th>
            <td>
                        2150.714
                         kWh
            </td>
</tr>,
          <tr>
            <th
                        scope="row"
            >
                        Electricity (normal)
            </th>
            <td>
                        2702.641
                         kWh
            </td>
</tr>,
          <tr>
            <th
                        scope="row"
            >
                        Gas
            </th>
            <td>
                        3781.676
                         m
                        <sup>
                                    3
                        </sup>
            </td>
</tr>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <th
                scope="row"
>
                Date recorded
</th>,
              <td>
                Friday, January 5th 2018
</td>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Date recorded",
                "scope": "row",
              },
              "ref": null,
              "rendered": "Date recorded",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Friday, January 5th 2018",
              },
              "ref": null,
              "rendered": "Friday, January 5th 2018",
              "type": "td",
            },
          ],
          "type": "tr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <th
                scope="row"
>
                Electricity (low)
</th>,
              <td>
                2150.714
                 kWh
</td>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Electricity (low)",
                "scope": "row",
              },
              "ref": null,
              "rendered": "Electricity (low)",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  2150.714,
                  " kWh",
                ],
              },
              "ref": null,
              "rendered": Array [
                2150.714,
                " kWh",
              ],
              "type": "td",
            },
          ],
          "type": "tr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <th
                scope="row"
>
                Electricity (normal)
</th>,
              <td>
                2702.641
                 kWh
</td>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Electricity (normal)",
                "scope": "row",
              },
              "ref": null,
              "rendered": "Electricity (normal)",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  2702.641,
                  " kWh",
                ],
              },
              "ref": null,
              "rendered": Array [
                2702.641,
                " kWh",
              ],
              "type": "td",
            },
          ],
          "type": "tr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <th
                scope="row"
>
                Gas
</th>,
              <td>
                3781.676
                 m
                <sup>
                                3
                </sup>
</td>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Gas",
                "scope": "row",
              },
              "ref": null,
              "rendered": "Gas",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  3781.676,
                  " m",
                  <sup>
                    3
</sup>,
                ],
              },
              "ref": null,
              "rendered": Array [
                3781.676,
                " m",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "3",
                  },
                  "ref": null,
                  "rendered": "3",
                  "type": "sup",
                },
              ],
              "type": "td",
            },
          ],
          "type": "tr",
        },
      ],
      "type": "tbody",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "borderless": true,
        "children": <tbody>
          <tr>
                    <th
                              scope="row"
                    >
                              Date recorded
                    </th>
                    <td>
                              Friday, January 5th 2018
                    </td>
          </tr>
          <tr>
                    <th
                              scope="row"
                    >
                              Electricity (low)
                    </th>
                    <td>
                              2150.714
                               kWh
                    </td>
          </tr>
          <tr>
                    <th
                              scope="row"
                    >
                              Electricity (normal)
                    </th>
                    <td>
                              2702.641
                               kWh
                    </td>
          </tr>
          <tr>
                    <th
                              scope="row"
                    >
                              Gas
                    </th>
                    <td>
                              3781.676
                               m
                              <sup>
                                        3
                              </sup>
                    </td>
          </tr>
</tbody>,
        "responsive": true,
        "responsiveTag": "div",
        "tag": "table",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <tr>
              <th
                            scope="row"
              >
                            Date recorded
              </th>
              <td>
                            Friday, January 5th 2018
              </td>
</tr>,
            <tr>
              <th
                            scope="row"
              >
                            Electricity (low)
              </th>
              <td>
                            2150.714
                             kWh
              </td>
</tr>,
            <tr>
              <th
                            scope="row"
              >
                            Electricity (normal)
              </th>
              <td>
                            2702.641
                             kWh
              </td>
</tr>,
            <tr>
              <th
                            scope="row"
              >
                            Gas
              </th>
              <td>
                            3781.676
                             m
                            <sup>
                                          3
                            </sup>
              </td>
</tr>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th
                  scope="row"
>
                  Date recorded
</th>,
                <td>
                  Friday, January 5th 2018
</td>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Date recorded",
                  "scope": "row",
                },
                "ref": null,
                "rendered": "Date recorded",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Friday, January 5th 2018",
                },
                "ref": null,
                "rendered": "Friday, January 5th 2018",
                "type": "td",
              },
            ],
            "type": "tr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th
                  scope="row"
>
                  Electricity (low)
</th>,
                <td>
                  2150.714
                   kWh
</td>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Electricity (low)",
                  "scope": "row",
                },
                "ref": null,
                "rendered": "Electricity (low)",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    2150.714,
                    " kWh",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  2150.714,
                  " kWh",
                ],
                "type": "td",
              },
            ],
            "type": "tr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th
                  scope="row"
>
                  Electricity (normal)
</th>,
                <td>
                  2702.641
                   kWh
</td>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Electricity (normal)",
                  "scope": "row",
                },
                "ref": null,
                "rendered": "Electricity (normal)",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    2702.641,
                    " kWh",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  2702.641,
                  " kWh",
                ],
                "type": "td",
              },
            ],
            "type": "tr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th
                  scope="row"
>
                  Gas
</th>,
                <td>
                  3781.676
                   m
                  <sup>
                                    3
                  </sup>
</td>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Gas",
                  "scope": "row",
                },
                "ref": null,
                "rendered": "Gas",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    3781.676,
                    " m",
                    <sup>
                      3
</sup>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  3781.676,
                  " m",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "3",
                    },
                    "ref": null,
                    "rendered": "3",
                    "type": "sup",
                  },
                ],
                "type": "td",
              },
            ],
            "type": "tr",
          },
        ],
        "type": "tbody",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
